xs/src/libslic3r/Model.hpp

```diff
@@ -28,221 +28,529 @@ typedef std::vector<ModelObject*> ModelObjectPtrs;
     TriangleMesh raw_mesh() const;
+
+    /// Get the raw bounding box
+    /// \return BoundingBoxf3
     BoundingBoxf3 raw_bounding_box() const;
+
+    /// Get the bounding box of the *transformed* given instance.
+    /// \param instance_idx size_t the index of the ModelInstance in the ModelInstance vector
+    /// \return BoundingBoxf3 the bounding box at the given index
     BoundingBoxf3 instance_bounding_box(size_t instance_idx) const;
+
+    /// Align the current ModelObject to ground by translating in the z axis the needed units.
     void align_to_ground();
+
+    /// Center the current ModelObject to origin.
     void center_around_origin();
+
+    /// Translate the current ModelObject with (x,y,z) units.
+    /// \param vector Vectorf3 the translation vector
     void translate(const Vectorf3 &vector);
+
+    /// Translate each TriangleMesh in every ModelVolume in this ModelObject.
+    /// \param x coordf_t the x units
+    /// \param y coordf_t the y units
+    /// \param z coordf_t the z units
     void translate(coordf_t x, coordf_t y, coordf_t z);
+
+    /// Scale the current ModelObject
+    /// \param factor float the scaling factor
     void scale(float factor);
+
+    /// Scale each TriangleMesh in every ModelVolume in this ModelObject.
+    /// \param versor Pointf3 the scaling factor in a 3d vector.
     void scale(const Pointf3 &versor);
+
+    /// Scale the current ModelObject to fit.
+    /// \param size Sizef3 the size vector
     void scale_to_fit(const Sizef3 &size);
+
+    /// Rotate the current ModelObject.
+    /// \param angle float the angle in radians
+    /// \param axis Axis the axis to be rotated around
     void rotate(float angle, const Axis &axis);
+
+    /// Mirror the current Model around a certain axis.
+    /// \param axis Axis enum member
     void mirror(const Axis &axis);
+
+    /// Transform the current ModelObject by a certain ModelInstance attributes.
+    /// \param instance ModelInstance the instance used to transform the current ModelObject
+    /// \param dont_translate bool whether to translate the current ModelObject or not
     void transform_by_instance(ModelInstance instance, bool dont_translate = false);
+
+    /// Get the number of the unique ModelMaterial objects in this ModelObject.
+    /// \return size_t the materials count
     size_t materials_count() const;
+
+    /// Get the number of the facets found in all ModelVolume objects in this ModelObject which are not modifier volumes.
+    /// \return size_t the facets count
     size_t facets_count() const;
+
+    /// Know whether there exists a TriangleMesh object that needs repair or not.
+    /// \return bool
     bool needed_repair() const;
+
+    /// Cut (Slice) the current ModelObject at a certain axis at a certain magnitude.
+    /// \param axis Axis the axis to slice at (X = 0 or Y or Z)
+    /// \param z coordf_t the point at the certain axis to cut(slice) the Model at
+    /// \param model the owner Model
```

Comment:
This is the Model which will get the resulting objects added. It can be the same model as the input one, or another one.